v2.1/gl/conversions.go

```diff
@@ -46,6 +48,13 @@ func PtrOffset(offset int) unsafe.Pointer {
 	return unsafe.Pointer(uintptr(offset))
 }
 
+// StrFree takes a Go string and allocates a C string in the C heap using malloc.
+// It is the caller's responsibility to free it by calling returned free.
+func StrFree(str string) (cstr *uint8, free func()) {
+	ptr := unsafe.Pointer(C.CString(str))
+	return (*uint8)(ptr), func() { C.free(ptr) }
+}
+
```

Comment:
I agree with the suggestion of making a helper that will assist with creating not just one, but multiple strings for use with funcs such as:

```Go
func CompileShaderIncludeARB(shader uint32, count int32, path **uint8, length *int32) {}
func CreateShaderProgramv(xtype uint32, count int32, strings **uint8) uint32 {}
func GetUniformIndices(program uint32, uniformCount int32, uniformNames **uint8, uniformIndices *uint32) {}
func TransformFeedbackVaryings(program uint32, count int32, varyings **uint8, bufferMode uint32) {}
```

How about something like this (discussed and pair-programmed with @slimsag):

```Go
// Strs takes a list of null-terminated Go strings and return's their C counterpart.
//
// The returned free function must be called once you are done using the strings in
// order to free the memory.
//
// If no strings are provided as a parameter, or if any string is not null-terminated,
// this function will panic.
func Strs(strs ...string) (cstrs **uint8, free func()) { ... }
```